frontend-wasm2/src/component/lower_imports.rs

```diff
@@ -0,0 +1,119 @@
+//! lowering the imports into the Miden ABI for the cross-context calls
+
+use std::{cell::RefCell, rc::Rc, str::FromStr};
```

Comment:
```suggestion
use alloc::rc::Rc;
use core::{cell::RefCell, str::FromStr};
```

In preparation for running the compiler in the browser, we should make sure to only reference libstd behind a `#[cfg(feature = "std")]` gate. Generally speaking though, if we can reference a type in libcore we should, failing that, liballoc.